contract/src/start-ibcToken.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file core eval script* to start the ibcToken contract.
+ *
+ * * see test-gimix-proposal.js to make a script from this file.
+ *
+ * The `permit` export specifies the corresponding permit.
+ */
+// @ts-check
+
+import { E } from '@endo/far';
+import { makeMarshal } from '@endo/marshal';
+
+const { Fail } = assert;
+
+const trace = (...args) => console.log('start-ibcToken', ...args);
+
+const fail = msg => {
+  throw Error(msg);
+};
+
+const evalConfig = {
+  tokenKeyword: 'TEST',
+  kitName: 'test',
+};
+
+/**
+ * confrims the provided key:
+ *  - starts with a letter or underscore (no leading numbers)
+ *  - only contains letters, numbers, or underscores without any whitespace
+ * @param {string} key
+ * @returns {boolean}
+ */
+export const isValidKey = key => /^[a-zA-Z_][a-zA-Z0-9_]*$/.test(key);
+assert(isValidKey(evalConfig.kitName), 'evalConfig.kitName is not valid');
+
+const BOARD_AUX = 'boardAux';
+const marshalData = makeMarshal(_val => Fail`data only`);
+
+/**
+ *
+ * @param {ERef<StorageNode>} chainStorage
+ * @param {string} boardId
+ * @returns {Promise<StorageNode>}
+ */
+const makeBoardAuxNode = async (chainStorage, boardId) => {
+  const boardAux = E(chainStorage).makeChildNode(BOARD_AUX);
+  return E(boardAux).makeChildNode(boardId);
+};
+
+/**
+ * @param {ERef<StorageNode>} chainStorage
+ * @param {ERef<import('@agoric/vats').Board>} board
+ * @param {ERef<Brand>} brand
+ */
+const publishBrandInfo = async (chainStorage, board, brand) => {
+  const [id, displayInfo, allegedName] = await Promise.all([
+    E(board).getId(brand),
+    E(brand).getDisplayInfo(),
+    E(brand).getAllegedName(),
+  ]);
+  const node = makeBoardAuxNode(chainStorage, id);
+  const aux = marshalData.toCapData(harden({ allegedName, displayInfo }));
+  void (await E(node).setValue(JSON.stringify(aux)));
+};
+
+/**
+ * @param {BootstrapPowers} powers
+ * @param {{ options?: {
+ *  ibcToken: {
+ *    bundleID: string;
+ *  };
+ *  tokenKeyword: string;
+ *  tokenDecimals: number;
+ *  proposedName: string;
+ *  denom: string;
+ * }}} config
+ */
+export const startIBCToken = async (powers, config) => {
```

Comment:
Aside from the `boardAux` stuff, this is in the SDK too: [publishInterchainAssetFromBank](https://github.com/Agoric/agoric-sdk/blob/mainnet1B-rc3/packages/inter-protocol/src/proposals/addAssetToVault.js#L55)